smithy-aws-apigateway-openapi/src/main/java/software/amazon/smithy/aws/apigateway/openapi/ApiGatewayDefault.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.apigateway.openapi;
+
+import software.amazon.smithy.openapi.OpenApiConfig;
+import software.amazon.smithy.utils.SetUtils;
+import software.amazon.smithy.utils.SmithyUnstableApi;
+
+public enum ApiGatewayDefault {
+    VERSION_2023_08_11("2023-08-11"),
+    DISABLED("disabled");
+
+    private final String version;
+
+    ApiGatewayDefault(String version) {
+        this.version = version;
+    }
+
+    @Override
+    public String toString() {
+        return version;
+    }
+
+    @SmithyUnstableApi
+    public OpenApiConfig setDefaults(OpenApiConfig config) {
```

Comment:
I think we should instead make this method abstract and implement in each enum value. I think a separate method for loading the right value from a string makes sense too (I think this is created using a NodeMapper already, so it just works with the toString implementation already too).